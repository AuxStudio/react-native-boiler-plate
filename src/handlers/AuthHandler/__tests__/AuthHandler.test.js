import React from 'react';
import renderer from 'react-test-renderer';

import { AuthHandler } from '..';

describe('AuthHandler', () => {
  const spies = [];
  const dispatch = jest.fn();

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<AuthHandler dispatch={dispatch} />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle getAuth', () => {
      const component = renderer.create(<AuthHandler dispatch={dispatch} />);
      const instance = component.getInstance();

      instance.getAuth();

      expect(dispatch).toHaveBeenCalled();
      expect(dispatch).toMatchSnapshot();
    });
  });

  describe('lifecycle methods', () => {
    it('should call getAuth in componentDidMount', () => {
      spies[0] = jest.spyOn(AuthHandler.prototype, 'getAuth');
      renderer.create(<AuthHandler dispatch={dispatch} />);

      expect(spies[0]).toHaveBeenCalled();
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
