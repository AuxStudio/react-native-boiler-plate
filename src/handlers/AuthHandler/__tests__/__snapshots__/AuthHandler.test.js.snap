// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHandler methods should handle getAuth 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "type": "getAuth",
      },
    ],
    Array [
      Object {
        "type": "getAuth",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`AuthHandler renders renders with minimum required props 1`] = `null`;
